File: fsharp-csharp-interop-demos/unit-testing-demo/Domain.Tests.FSharp/DalekTests.fs
module DalekTests

open System
open FsUnit.Xunit
open Xunit
open Domain

[<Fact>]
let ``Dalek creation via ctor returns valid result`` () =
    // Calling a C# Ctor with nullable props from F#:
    let dalek = new Dalek(Nullable 1, Nullable 10)

    dalek.Id |> should equal 1
    dalek.Power |> should equal 10

[<Fact>]
let ``Dalek creation via props returns valid result`` () =
    // Using C# object initializer (with nullable props) syntax from F#:
    let dalek = new Dalek(Id = Nullable 1, Power = Nullable 10)

    dalek.Id |> should equal 1
    dalek.Power |> should equal 10

File: fsharp-csharp-interop-demos/unit-testing-demo/Domain.Tests.CSharp/DalekTests.cs
using FluentAssertions;
using Xunit;

namespace Domain.Tests.CSharp
{
    public class DalekTests
    {
        [Fact]
        public void Dalek_creation_via_ctor_returns_valid_result()
        {
            var dalek = new Dalek(1, 10);

            dalek.Id.Should().Be(1);
            dalek.Power.Should().Be(10);
        }

        [Fact]
        public void Dalek_creation_via_props_returns_valid_result()
        {
            var dalek = new Dalek
            {
                Id = 1,
                Power = 10
            };

            dalek.Id.Should().Be(1);
            dalek.Power.Should().Be(10);
        }
    }
}
